components/brave_shields/browser/ad_block_service_helper.h

```diff
@@ -25,6 +25,9 @@ std::vector<adblock::FilterList>::const_iterator FindAdBlockFilterListByLocale(
 std::vector<adblock::FilterList> RegionalCatalogFromJSON(
     const std::string& catalog_json);
 
+void MergeCspDirectiveInto(base::Optional<std::string> from,
```

Comment:
personally I don't see much sense in using Optional here and in the service (also CC recommends against it https://chromium.googlesource.com/chromium/src/+/master/docs/optional.md#when-not-to-use).
IMO just (potentially, empty) strings would be more readable.

Reply:
I think I disagree here. CSP is a security-sensitive topic, and it's important to make sure these directives are handled with appropriate care. Having the `Optional` enforces a stronger contract than just passing strings around, with a distinct semantic difference between "there is a directive here" and "no directive". I think it'd be tempting to bypass the merge function here altogether if working directly with strings; we have already run into some issues with incorrectly concatenated CSP directives in the past (e.g. https://github.com/brave/brave-browser/issues/14752).